dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task04305UpdateWorkflowActionTable.java

```diff
@@ -0,0 +1,596 @@
+     * @return
+     */
+    private String updateSchemeIdsForActions() {
+        if (DbConnectionFactory.isMySql()) {
+            return MYSQL_UPDATE_SCHEME_IDS_FOR_ACTIONS;
+        } else if (DbConnectionFactory.isPostgres()) {
+            return POSTGRES_UPDATE_SCHEME_IDS_FOR_ACTIONS;
+        } else if (DbConnectionFactory.isMsSql()) {
+            return MSSQL_UPDATE_SCHEME_IDS_FOR_ACTIONS;
+        } else if (DbConnectionFactory.isOracle()) {
+            return ORACLE_UPDATE_SCHEME_IDS_FOR_ACTIONS;
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Retrieves the relationships between workflow actions and the workflow scheme where they live.
+     * @return
+     */
+    private String selectSchemeIdsForActions() {
+        if (DbConnectionFactory.isMySql()) {
+            return MYSQL_SELECT_SCHEME_IDS_FOR_ACTIONS;
+        } else if (DbConnectionFactory.isPostgres()) {
+            return POSTGRES_SELECT_SCHEME_IDS_FOR_ACTIONS;
+        } else if (DbConnectionFactory.isMsSql()) {
+            return MSSQL_SELECT_SCHEME_IDS_FOR_ACTIONS;
+        } else if (DbConnectionFactory.isOracle()) {
+            return ORACLE_SELECT_SCHEME_IDS_FOR_ACTIONS;
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Inserts the relationships between workflow actions and the workflow steps where they can
+     * be used in the new intermediate table.
+     * @return
+     */
+    private String insertActionsAndSteps() {
+        if (DbConnectionFactory.isMySql()) {
+            return MYSQL_INSERT_INTO_INTERMEDIATE_TABLE;
+        } else if (DbConnectionFactory.isPostgres()) {
+            return POSTGRES_INSERT_INTO_INTERMEDIATE_TABLE;
+        } else if (DbConnectionFactory.isMsSql()) {
+            return MSSQL_INSERT_INTO_INTERMEDIATE_TABLE;
+        } else if (DbConnectionFactory.isOracle()) {
+            return ORACLE_INSERT_INTO_INTERMEDIATE_TABLE;
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Retrieves the relationships between workflow actions and the workflow steps where they can
+     * be used.
+     * @return
+     */
+    private String selectActionsAndSteps() {
+        if (DbConnectionFactory.isMySql()) {
+            return MYSQL_SELECT_ACTIONS_AND_STEPS;
+        } else if (DbConnectionFactory.isPostgres()) {
+            return POSTGRES_SELECT_ACTIONS_AND_STEPS;
+        } else if (DbConnectionFactory.isMsSql()) {
+            return MSSQL_SELECT_ACTIONS_AND_STEPS;
+        } else if (DbConnectionFactory.isOracle()) {
+            return ORACLE_SELECT_ACTIONS_AND_STEPS;
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Adds the new {@code scheme_id} column to the {@code workflow_action} table so that it is now
+     * related directly with a workflow scheme.
+     * @return
+     */
+    private String addSchemeIdColumn() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 5, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1142)